### PR TITLE
Bug 1732929: Pin dependencies to 4.1

### PR DESCRIPTION
**Pull request by vikaschoudhary16**
Fixing:

- https://bugzilla.redhat.com/show_bug.cgi?id=1713061 (Missing node prevents machine from being delete)
- https://bugzilla.redhat.com/show_bug.cgi?id=1732929 (4.1.4 cluster creating 400+ eviction requests a second)
- Pinning dependencies, cluster-api and cluster-api-actuator-pkg,  to 4.1 release

Will rebase this PR once https://github.com/openshift/cluster-api-actuator-pkg/pull/111 gets merged. This PR is further updating cluster-api-pkg-dependencies to 4.1 release

### CODE REVIEW

File: vendor/github.com/openshift/api/config/v1/types_infrastructure.go [hunk 1]
```diff
@@ -85,8 +92,113 @@ const (
...
+	// +optional
+	AWS *AWSPlatformStatus `json:"aws,omitempty"`
+
+	// Azure contains settings specific to the Azure infrastructure provider.
+	// +optional
+	Azure *AzurePlatformStatus `json:"azure,omitempty"`
+
+	// GCP contains settings specific to the Google Cloud Platform infrastructure provider.
+	// +optional
+	GCP *GCPPlatformStatus `json:"gcp,omitempty"`
```
Comment (michaelgugino) [inline vendor/github.com/openshift/api/config/v1/types_infrastructure.go]: This looks like 4.2 stuff.

**Comment by eparis** [comment]
We MUST have a 4.1.z target release bug in the title (of the form `Bug NNNNNNN: title title title`)
That bug MUST 'depend on' an 4.2 target release bz.
And that 4.2 BZ MUST be VERIFIED by QE.
We also need lgtm and approved from the team after review.

**Comment by vikaschoudhary16** [comment]
> We MUST have a 4.1.z target release bug in the title (of the form `Bug NNNNNNN: title title title`)
> That bug MUST 'depend on' an 4.2 target release bz.
> And that 4.2 BZ MUST be VERIFIED by QE.
> We also need lgtm and approved from the team after review.

Updated. 4.2 bug is "ON_QA" so will wait till that gets "VERIFIED".

**Comment by enxebre** [comment]
cc @mfojtik

**Comment by enxebre** [comment]
@eparis @vikaschoudhary16 this lgtm, can we ship it?

**Comment by knobunc** [comment]
@vikaschoudhary16 I'm pulling approval for the backport until the tests pass.  I tried a few times and I think it's a problem with the PR.

**Comment by ingvagabund** [comment]
From https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_cluster-api-provider-aws/253/pull-ci-openshift-cluster-api-provider-aws-release-4.1-e2e-aws-operator/111/artifacts/e2e-aws-operator/pods/openshift-machine-api_machine-api-controllers-579b8c845b-427sv_machine-controller.log:

```
I0917 09:36:16.493790       1 controller.go:227] Machine "machine1" deletion successful
```

From https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_cluster-api-provider-aws/253/pull-ci-openshift-cluster-api-provider-aws-release-4.1-e2e-aws-operator/111/artifacts/e2e-aws-operator/container-logs/test.log:

```
E0917 09:35:39.750590    6102 infra.go:464] Machine "machine1" not yet deleted
I0917 09:35:39.750938    6102 infra.go:365] Deleting object "pdb"
```

Machine `machine1` was deleted eventually, it just happened 37s after the timeout.

**Pull request merged**